File: CarAgency/ClientApp/src/domain/models/direcciones/Direccion.ts
import IDireccion from "domain/interfaces/direcciones/IDireccion";
import Localidad from "domain/models/direcciones/Localidad";
import Pais from "domain/models/direcciones/Pais";
import Partido from "domain/models/direcciones/Partido";
import Provincia from "domain/models/direcciones/Provincia";

export default class Direccion implements IDireccion {
    constructor(){
    this.DireccionId = 0;
    this.Pais = new Pais();
    this.Provincia = new Provincia();
    this.Partido = new Partido();
    this.Localidad = new Localidad();
    this.Calle = "";
    this.NumeroCalle = "";
    this.CodigoPostal = "";
        
    }
    DireccionId: number;
    Pais: Pais;
    Provincia: Provincia;
    Partido: Partido;
    Localidad: Localidad;
    Calle: string;
    NumeroCalle: string;
    CodigoPostal: string;

}

File: Domain/DTO/Direcciones/Direccion.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Domain.DTO.Direcciones
{
    public class Direccion
    {
        [Key]
        public int DireccionId { get; set; }
        public Pais Pais { get; set; }
        public Provincia Provincia { get; set; }
        public Partido Partido { get; set; }
        public Localidad Localidad { get; set; }

        [Column(TypeName = "nvarchar(100)")]
        public string Calle { get; set; }

        public Int16 NumeroCalle { get; set; }

        [Column(TypeName = "nvarchar(10)")]
        public string CodigoPostal { get; set; }
    }
}
